app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/model/AdminVerification.kt

```diff
@@ -0,0 +1,76 @@
+package de.tum.`in`.tca.ticketcheck.component.ticket.model
+
+import android.content.Context
+import de.tum.`in`.tca.ticketcheck.api.AuthenticationManager
+import de.tum.`in`.tca.ticketcheck.api.exception.NoPrivateKey
+import de.tum.`in`.tca.ticketcheck.utils.Const
+import de.tum.`in`.tca.ticketcheck.utils.Utils
+import java.io.IOException
+import java.math.BigInteger
+import java.security.SecureRandom
+import java.util.*
+
+data class AdminVerification(var signature: String = "",
+                             var date: String = "",
+                             var rand: String = "",
+                             var data: Any? = null) {
+    companion object {
+        @Throws(NoPrivateKey::class)
+        fun getAdminVerification(c: Context): AdminVerification {
+            setKeys(c)
+            //Create some data
+            val date = Date().toString()
+            val rand = BigInteger(130, SecureRandom()).toString(32)
+
+            //Sign this data for verification
+            val am = AuthenticationManager(c)
+            val signature = am.sign(date + rand)
+
+            return AdminVerification(
+                    signature = signature,
+                    date = date,
+                    rand = rand,
+                    data = null
+            )
+        }
+
+        @Throws(NoPrivateKey::class)
+        fun getAdminVerification(c: Context, data: Any?): AdminVerification {
+            val ret = this.getAdminVerification(c)
+            ret.data = data
+            return ret
+        }
+
+        @Throws(IOException::class)
+        fun createAdminVerification(context: Context, data: Any?): AdminVerification {
+            var adminVerification: AdminVerification? = null
+            try {
+                adminVerification = AdminVerification.getAdminVerification(context, data)
+            } catch (noPrivateKey: NoPrivateKey) {
+                noPrivateKey.printStackTrace()
+                throw IOException()
+            }
+
+            return adminVerification
+        }
+
+        fun setKeys(context: Context) {
+            Utils.setSetting(context, Const.PUBLIC_KEY, "MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDnQmMnWxKL3kMd6a5MWC7sY8DBG7YkLVdfgaxr\n" +
+                    "XIE9Y25bbsSmq/1zfKLlph3j3wXm94mmxeH9rG92BcBe1MCeT4rAdYiQLjvJltk6yekZz9O4XDnR\n" +
+                    "8rvVu+9HyJVHm/ysipOeKHJreFzOxlQ2m1FWZ1djtiCEzHKEdWTQjzpc4QIDAQAB\n")
+            Utils.setSetting(context, Const.PRIVATE_KEY, "MIICdwIBADANBgkqhkiG9w0BAQEFAASCAmEwggJdAgEAAoGBAOdCYydbEoveQx3prkxYLuxjwMEb\n" +
```

Comment:
Guess this is here for testing? You should generate a new KeyPair on startup, that gets uploaded/registered on the server and then can be unlocked by an admin.

Reply:
Ok, I'll then create a new endpoint on the backend to upload the key to.